pkg/api/validation/validation_test.go

```diff
@@ -228,6 +245,29 @@ func TestValidatePorts(t *testing.T) {
 			if errs[i].(*errors.ValidationError).Field != v.F {
 				t.Errorf("%s: expected errors to have field %s: %v", k, v.F, errs[i])
 			}
+			detail := errs[i].(*errors.ValidationError).Detail
+			switch k {
```

Comment:
Rather than switching on the key, add your expectation to the test structure and check it here.